src/state/selection/types.ts

```diff
@@ -76,3 +76,8 @@ export interface SetRecentColorsAction {
     payload: string[];
     type: string;
 }
+
+export interface SetSelectedAgentAction {
+    payload: number;
```

Comment:
is this type right? isn't the payload either an empty object or an object with all the metadata?

Reply:
You're right, it should be `AgentMetadata`. Changing this, and changing everything to be `selectedAgentMetadata` and `setSelectedAgentMetadata` instead of `selectedAgent` and `setSelectedAgent`